taskcluster/docker/node/build.py

```diff
@@ -150,20 +189,35 @@ def check_manifest(path, buildid_version):
                 continue
             _found = True
             if not _id.endswith(ID_ALLOWLIST):
-                raise Exception(f"{_key}.gecko.id {_id} must end with one of the following suffixes!\n{ID_ALLOWLIST}")
+                raise Exception(
+                    f"{_key}.gecko.id {_id} must end with one of the following suffixes!\n{ID_ALLOWLIST}"
+                )
             else:
                 print(f"Add-on id {_id} matches the allowlist.")
         if manifest["version"] != buildid_version:
-            raise Exception(f"{manifest['version']} doesn't match buildid version {buildid_version}!")
+            raise Exception(
+                f"{manifest['version']} doesn't match buildid version {buildid_version}!"
+            )
+        if manifest["manifest_version"] == 3 and not is_version_mv3_compliant(manifest["version"]):
```

Comment:
Seeing the check against the `buildid_version` right before, this new check would mean we somehow failed to make a compliant version string, right?

Reply:
Yeah, that's right. This is checking the generated versions in V3 manifest(s) to make sure it compliant. It is more likely that it will fail because of the manually crafted part of the string. But the in-container docker script/executable is not unit tested. So I added this as a sanity check/sign-post.